elf_reader_test.go

```diff
@@ -314,32 +314,58 @@ func TestLibBPFCompat(t *testing.T) {
 	})
 }
 
+type elfSectionProgType struct {
```

Comment:
I'd prefer to have this type declared in `TestGetProgType`, since it has little meaning outside of that scope. It can be named something generic like `testCase` to avoid it popping up in greps, and the uppercasing of the members is also not necessary as far as I can see.

Reply:
fixed